packages/common/src/common.ts

```diff
@@ -636,67 +702,45 @@ export class Common extends EventEmitter {
     return BigInt(ttd)
   }
 
-  /**
-   * True if block number provided is the hardfork (given or set) change block
-   * @param blockNumber Number of the block to check
-   * @param hardfork Hardfork name, optional if HF set
-   * @returns True if blockNumber is HF block
-   */
-  isHardforkBlock(blockNumber: BigIntLike, hardfork?: string | Hardfork): boolean {
-    blockNumber = toType(blockNumber, TypeOutput.BigInt)
-    hardfork = hardfork ?? this._hardfork
-    const block = this.hardforkBlock(hardfork)
-    return typeof block === 'bigint' && block !== BigInt(0) ? block === blockNumber : false
-  }
-
   /**
    * Returns the change block for the next hardfork after the hardfork provided or set
    * @param hardfork Hardfork name, optional if HF set
-   * @returns Block number or null if not available
+   * @returns Block timestamp, number or null if not available
    */
   nextHardforkBlock(hardfork?: string | Hardfork): bigint | null {
     hardfork = hardfork ?? this._hardfork
-    let hfBlock = this.hardforkBlock(hardfork)
-    // If this is a merge hardfork with block not set, then we fallback to previous hardfork
-    // to find the nextHardforkBlock
-    if (hfBlock === null && hardfork === Hardfork.Merge) {
-      const hfs = this.hardforks()
-      const mergeIndex = hfs.findIndex((hf) => hf.ttd !== null && hf.ttd !== undefined)
-      if (mergeIndex < 0) {
-        throw Error(`Merge hardfork should have been found`)
-      }
-      hfBlock = this.hardforkBlock(hfs[mergeIndex - 1].name)
-    }
-    if (hfBlock === null) {
+    const hfs = this.hardforks()
+    let hfIndex = hfs.findIndex((hf) => hf.name === hardfork)
+    // If the current hardfork is merge go, one behind as merge hf is not part of these
```

Comment:
The meaning of this comment isn't clear to me.  Can you clarify?

Reply:
the comment means is to ignore merge hardfork to determine what is current fork and what is the next, its a rehash of what was previously being done as `hfBlock = this.hardforkBlock(hfs[mergeIndex - 1].name`